src/control/system/database_test.go

```diff
@@ -74,6 +74,52 @@ func waitForLeadership(t *testing.T, ctx context.Context, db *Database, gained b
 	}
 }
 
+func TestSystem_Database_filterMembers(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	db := MockDatabase(t, log)
+	memberStates := []MemberState{
+		MemberStateUnknown, MemberStateAwaitFormat, MemberStateStarting,
+		MemberStateReady, MemberStateJoined, MemberStateStopping, MemberStateStopped,
+		MemberStateEvicted, MemberStateErrored, MemberStateUnresponsive,
+	}
+
+	for i, ms := range memberStates {
+		if err := db.AddMember(MockMember(t, uint32(i), ms)); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	for name, tf := range map[string]func(t *testing.T){
+		"individual state filters": func(t *testing.T) {
+			for _, ms := range memberStates {
+				matchLen := len(db.filterMembers(ms))
```

Comment:
Don't we also want to be sure that it's returning the correct members, not just the right number?

Reply:
Good call, fixed.